### DISCUSSION TITLE
Deployment don't work anymore without a having change anything

### DISCUSSION THREAD

**Discussion by YsarocK** [vercel/community#998] [Help] [answered]
Hi,

I have two Nuxt 2 project running on Vercel. The two have quite-the-same config.
I made changes on them, but really dumb things such as console.log or... no dependency or logic change.

The build process is not the same, used yarn before and uses npm now.

It causes every deploy to show "The Serverless Function "index" is 256.75mb which exceeds the maximum size limit of 50mb." error.

I really do not understand, I have 10+ running project on Vercel, i'm blocked by those one.

**Comment by amyegan** [comment, marked as answer]
Sorry you're running into errors @YsarocK. The [Install Command](https://vercel.com/docs/concepts/deployments/configure-a-build#install-command) is automatically selected based on the presence, or absence, of a specific type of lockfile. A `pnpm-lock.yaml` will make your project automatically choose `pnpm`, a `package-lock.json` will make your project automatically use `npm`, and `yarn` will be used if neither of the other lockfiles are found. You can manually override the Install Command in your [Project Settings](https://vercel.com/docs/concepts/projects/overview#build-&-development-settings-and-root-directory) if you want to ensure that `yarn install` is always used.

If you are not using a lockfile and don't have specific version numbers pinned in your `package.json` file, it is possible that some new versions of your dependencies are making the Serverless Function too large. You can use tools [Bundlephobia](https://bundlephobia.com/) or [bundlejs](https://bundlejs.com/) to identify large dependencies. Do you get the same error if you try to redeploy an older successful deployment?

**Reply by YsarocK** [reply]
Thanks for reaching out. I have a package-lock, but last working deploys logs show "using yarn". The last failing ones use npm and all fail. I have tried overhiding install command by adding yarn install, but It's not considered.

**Reply by amyegan** [reply]
Can you share the build IDs of your last working deploy and a recent one that is failing? That will let me look up the logs. You can find the ID in the URL of the deployment overview page in your project dashboard. For example, `vercel.com/my-username/my-project/6PtVvr3UofEWZNt3X6AY5JgdNf3s` has the ID `6PtVvr3UofEWZNt3X6AY5JgdNf3s`

**Reply by YsarocK** [reply]
- Working one : FcaXE2JfqYXg8G7eoJG9DznNrprZ
- Failed one (first of them) : G6tjBfqEKzSmbXGYsQyNW1YUe9c1

Thanks a lot for the help Amy, have a good weekend.

**Reply by amyegan** [reply]
Thanks for sharing these. I see a config warning in the logs. It looks like you're using the legacy [`builds`](https://vercel.com/docs/project-configuration#legacy/builds) config, so the Build and Development Settings changes you make in the Dashboard will have no effect. If you set the [`installCommand`](https://vercel.com/docs/project-configuration#project-configuration/install-command) in your `vercel.json` configuration file to force it to use `yarn install`, does that resolve the error? 

I'm also seeing some warnings about various packages for the failing build, but those warnings are not present for the successful build. Maybe some of the packages have had updates since then. You could try setting the packages to older versions (matching the last successful build or whichever versions are working in your local environment) in your `package.json`.

**Reply by YsarocK** [reply]
Hi Amy, 

Forcing the yarn usage & deleting the package-lock worked ! Thanks a lot for your help.

Have good day :)
